flink-table-store-core/src/main/java/org/apache/flink/table/store/file/FileStoreImpl.java

```diff
@@ -201,17 +204,28 @@ public static FileStoreImpl createWithPrimaryKey(
                                 .collect(Collectors.toList()));
 
         MergeFunction mergeFunction;
+        Map<String, String> rightConfMap =
+                options.getFilterConf(e -> e.getKey().endsWith(".aggregate-function"));
         switch (mergeEngine) {
             case DEDUPLICATE:
                 mergeFunction = new DeduplicateMergeFunction();
                 break;
             case PARTIAL_UPDATE:
-                List<LogicalType> fieldTypes = rowType.getChildren();
-                RowData.FieldGetter[] fieldGetters = new RowData.FieldGetter[fieldTypes.size()];
-                for (int i = 0; i < fieldTypes.size(); i++) {
-                    fieldGetters[i] = RowData.createFieldGetter(fieldTypes.get(i), i);
+                mergeFunction = new PartialUpdateMergeFunction(rowType);
+                break;
+            case AGGREGATION:
+                Set<String> valueSet = new HashSet<>(rightConfMap.values());
+                if (valueSet.size() != 1 || !valueSet.contains("sum")) {
+                    throw new IllegalArgumentException(
+                            "Aggregate function must be the same for all columns");
```

Comment:
Use a factory class to produce corresponding aggregate function. If we'd like to support other aggregate functions in the future this implementation is not friendly for changes.